version: '2.2'
services:
  nsd_authoritative:
    build:
      dockerfile: Dockerfile
      context: nsd
    container_name: nsd_in
    hostname: nsd_authoritative
    ports:
      - "53:53"
      - "53:53/udp"
    volumes:
      - ./etc/nsd/conf:/etc/nsd
      - ./etc/nsd/zones:/zones
      - nsd_db:/var/db/nsd
    healthcheck:
      test: ["CMD", "nsd-control", "status"]
      timeout: 10s
      retries: 5



  knot_authoritative:
    image: cznic/knot:2.7
    container_name: knot_out
    hostname: knot_authoritative
    volumes:
      - ./etc/knot/storage:/storage
      - ./etc/knot/config:/config
      - ./etc/knot/rundir:/rundir
    command: /bin/bash -c "knotd -c /config/knot.conf"
    ports:
      - "530:53"
      - "530:53/udp"
    healthcheck:
      test: ["CMD", "knotc", "status"]
      timeout: 10s
      retries: 5
    depends_on:
      nsd_authoritative:
        condition: service_healthy

  dns_api:
    build:
      dockerfile: Dockerfile
      context: dnspython
    container_name: dns_api
    hostname: dns_api
    volumes:
      - ./dnspython/app:/usr/src/app
      - ./dnsviz/data:/usr/src/app/data
    depends_on:
#      dnsviz:
      knot_authoritative:
        condition: service_healthy

  # dnsviz:
  #   image: dnsviz/dnsviz
  #   container_name: dnsviz
  #   hostname: dns_viz
  #   volumes:
  #     - ./dnsviz/data:/data:rw
  #   depends_on:
  #     knot_authoritative:
  #       condition: service_healthy
  #   command: [ "/usr/bin/python3", "-u","/data/app.py" ]
  #   entrypoint: [""]
  #   healthcheck:
  #     test: ["CMD", "dnsviz", "help"]
  #     timeout: 10s
  #     retries: 5

volumes:
  nsd_db:
